src/zenml/client.py

```diff
@@ -2510,30 +2442,69 @@ def list_pipelines(
         )
 
     def get_pipeline(
-        self, name_id_or_prefix: Union[str, UUID]
+        self,
+        id: Optional[UUID] = None,
+        name: Optional[str] = None,
+        version: Optional[str] = None,
     ) -> PipelineResponseModel:
         """Get a pipeline by name, id or prefix.
 
         Args:
-            name_id_or_prefix: The name, id or prefix of the pipeline.
+            id: The id of the pipeline.
+            name: The name of the pipeline.
+            version: The pipeline version. If left empty, will return
+                the latest version.
 
         Returns:
             The pipeline.
+
+        Raises:
+            ValueError: If both name and ID are unset.
+            KeyError: If no pipelines were found for the given name and version.
         """
-        return self._get_entity_by_id_or_name_or_prefix(
-            get_method=self.zen_store.get_pipeline,
-            list_method=self.list_pipelines,
-            name_id_or_prefix=name_id_or_prefix,
-        )
+        if id and (name or version):
+            logger.warning(
+                "You specified both an ID as well as name/version of the "
+                "pipeline to get. Ignoring name and version and fetching the "
+                "pipeline by ID."
+            )
+            name = None
+            version = None
+        if not (id or name):
+            raise ValueError(
+                "Please specify either the ID or the name of the pipeline to "
+                "get."
+            )
 
-    def delete_pipeline(self, name_id_or_prefix: Union[str, UUID]) -> None:
+        if id:
+            return self.zen_store.get_pipeline(id)
+        else:
+            pipelines = self.list_pipelines(
+                size=1, sort_by="desc:created", name=name, version=version
```

Comment:
Ah, I was assuming the ZenStore had some logic to only return the latest version by default but I didn't find anything like that. In that case, there's still a piece missing here I believe.

Reply:
You're looking at the `get_pipeline` method here which should only return one pipeline I'm assuming.

My intention when implementing this:
- If an ID is provided, ignore the rest and return pipeline with that ID
- If name and version are provided, get the pipeline matching those two
- If only name is provided, return latest version for that pipeline

Maybe you can check again but I think the logic looks correct, or would you expect different behaviour?

Reply:
`list_pipelines` IMO should actually return all versions if the filter includes a name but no version?

Reply:
yes yes this is fine, I thought `get_pipeline` had a flaw when being called with only a name but since you return the latest pipeline of that name you should correctly get the latest version